casper/src/test/resources/PoSTest/test_bonding_succeeds.rho

```diff
@@ -0,0 +1,70 @@
+            posRevAddressCh in {
+          revAddressOps!("fromDeployerId", *deployerId, *deployerRevAddressCh) |
+          revAddressOps!("fromPublicKey", posPk, *posRevAddressCh) |
+          for(@deployerRevAddress <- deployerRevAddressCh; @posRevAddress <- posRevAddressCh){
+            new deployerVaultCh, posVaultCh in {
+              @RevVault!("findOrCreate", deployerRevAddress, *deployerVaultCh) |
+              @RevVault!("findOrCreate", posRevAddress, *posVaultCh) |
+              for(@(true, deployerVault) <- deployerVaultCh;
+                  @(true, posVault) <- posVaultCh) {
+                new poSBalanceCh, deployerBalanceCh in {
+                  @posVault!("balance", *poSBalanceCh) |
+                  @deployerVault!("balance", *deployerBalanceCh) |
+                  for(@initialPoSBalance <- poSBalanceCh;
+                      @initialDeployerBalance <- deployerBalanceCh){
+                    new PoSCh in {
+                      rl!(`rho:rchain:pos`, *PoSCh) |
+                      for(@(_, PoS) <- PoSCh){
+                        new bondAck in {
+                          @PoS!("bond", 100, *bondAck) |
+                          for(@(true, _) <- bondAck){
+                            new bondsCh,
+                                rewardsCh,
+                                deployerIdOps(`rho:rchain:deployerId:ops`),
+                                deployerPkCh in {
+                              @PoS!("getBonds", *bondsCh) |
+                              @posVault!("balance", *poSBalanceCh) |
+                              @deployerVault!("balance", *deployerBalanceCh) |
+                              @PoS!("getRewards", *rewardsCh) |
+                              deployerIdOps!("pubKeyBytes", *deployerId, *deployerPkCh) |
+                              for(@finalPoSBalance <- poSBalanceCh;
+                                  @finalDeployerBalance <- deployerBalanceCh;
+                                  @bonds <- bondsCh;
+                                  @rewards <- rewardsCh;
+                                  @deployerPk <- deployerPkCh){
+                                match bonds.get(deployerPk) {
+                                  bondAmount => {
+                                    new assert(`rho:test:assertAck`), testSuiteCompleted(`rho:test:testSuiteCompleted`), ack in {
+                                      testSuiteCompleted!(true) |
+                                      assert!("add the new bond to the bonds map", 1, bondAmount == 100, "", *ack) |
+                                      assert!("decrease deployer vault balance by 100", 1, initialDeployerBalance - finalDeployerBalance == 100, "", *ack) |
+                                      assert!("increase PoS vault balance by 100", 1, finalPoSBalance - initialPoSBalance == 100, "", *ack) |
+                                      assert!("record no rewards for new validator", 1, rewards.get(deployerPk) == Nil, "", *ack)
```

Comment:
Since there are multiple messages sent to the ack channel. The result collector doesn't know when the assertions have finished. 

The original idea was that testSuiteCompleted !(true) happens at the end of the assertions to notify the test result collector that this test is finished

Reply:
Yeah. Using `testSuiteCompleted` here is definitely a hack. I'm weighing that against the syntactic overhead of using `RhoSpec` to run these assertions. I don't see a middle-ground between removing the "has finished" assertion in `RhoSpec` and using `testSuiteCompleted` here, but I'm certainly open to ideas.

Reply:
I think we should have an assertMany construct either as a system process or a rholang wrapper on top of the assert system process